File: ios/CircularBuffer/CircularBufferErrors.swift
import Foundation

/// Error codes surfaced to JS via the Promise rejection code.
enum CircularBufferErrorCode {
    static let cameraBusy = "E_CAMERA_BUSY"
    static let cameraPermission = "E_CAMERA_PERMISSION"
    static let encoderInitFailed = "E_ENCODER_INIT_FAILED"
    static let notBuffering = "E_NOT_BUFFERING"
    static let alreadyBuffering = "E_ALREADY_BUFFERING"
    static let storageFull = "E_STORAGE_FULL"
    static let saveFailed = "E_SAVE_FAILED"
}

struct CircularBufferError: Error {
    let code: String
    let message: String
}

File: android/app/src/main/java/com/voicecambuffer/circularbuffer/CircularBufferErrors.kt
package com.voicecambuffer.circularbuffer

/** Error codes surfaced to JS via Promise.reject(code, message). */
object CircularBufferErrorCodes {
    const val CAMERA_BUSY = "E_CAMERA_BUSY"
    const val CAMERA_PERMISSION = "E_CAMERA_PERMISSION"
    const val ENCODER_INIT_FAILED = "E_ENCODER_INIT_FAILED"
    const val NOT_BUFFERING = "E_NOT_BUFFERING"
    const val ALREADY_BUFFERING = "E_ALREADY_BUFFERING"
    const val STORAGE_FULL = "E_STORAGE_FULL"
    const val SAVE_FAILED = "E_SAVE_FAILED"
    const val NO_ACTIVITY = "E_NO_ACTIVITY"
}

class CameraBusyException(message: String) : Exception(message)
class EncoderInitException(message: String, cause: Throwable? = null) : Exception(message, cause)
